core/trino-main/src/main/java/io/trino/sql/planner/plan/SimplePlanRewriter.java

```diff
@@ -69,11 +68,9 @@ public PlanNode defaultRewrite(PlanNode node)
          */
         public PlanNode defaultRewrite(PlanNode node, C context)
         {
-            List<PlanNode> children = node.getSources().stream()
-                    .map(child -> rewrite(child, context))
-                    .collect(toImmutableList());
-
-            return replaceChildren(node, children);
+            ImmutableList.Builder<PlanNode> children = ImmutableList.builderWithExpectedSize(node.getSources().size());
```

Comment:
Can you move it to separate PR? I'd like to merge this.

Reply:
https://github.com/trinodb/trino/pull/11919